// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.power_bookmarks";

option optimize_for = LITE_RUNTIME;

package power_bookmarks;

// PowerType is persisted to disk and therefore shouldn't be reoderered or
// changed.
enum PowerType {
  // All powers should have type so this should be unused.
  POWER_TYPE_UNSPECIFIED = 0;
  // Used for testing.
  POWER_TYPE_MOCK = 1;
}

message PowerSpecifics {
  optional string placeholder = 1;
}

message PowerBookmarkSpecifics {
  // The primary key for local storage.
  /* required */ optional string guid = 1;

  // Powers associated with a URL.
  /* required */ optional string url = 2;

  // Tracks the type of the power.
  /* required */ optional PowerType power_type = 3;

  // Tracks when this was added.
  /* required */ optional int64 creation_time_usec = 4;

  // Tracks when this was last modified.
  /* required */ optional int64 update_time_usec = 5;

  // Power specifics.
  /* required */ optional PowerSpecifics power_specifics = 100;
}
